resolver-dns/src/main/java/io/netty/resolver/dns/DefaultDnsCache.java

```diff
@@ -301,21 +317,53 @@ void add(DefaultDnsCacheEntry e) {
                                 assert replacedEntry == null;
                                 replacedEntry = entry;
                             }
-                        }
+                            i++;
+                        } while (i < entries.size());
```

Comment:
consider moving the post-increment above to a pre-increment here for more compactness `i` -> `++i`